Demo/Common/IntegrationExamples/Components/ApplePayComponentAdvancedFlowExample.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Copyright (c) 2023 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import AdyenActions
+import AdyenComponents
+import PassKit
+
+internal final class ApplePayComponentAdvancedFlowExample: InitialDataAdvancedFlowProtocol {
+
+    // MARK: - Properties
+
+    internal var paymentMethods: PaymentMethods?
+    internal var applePayComponent: ApplePayComponent?
+    internal weak var presenter: PresenterExampleProtocol?
+
+    // MARK: - Initializers
+
+    internal init() {}
+
+    internal func start() {
+        presenter?.showLoadingIndicator()
+        requestPaymentMethods(order: nil) { [weak self] result in
+            guard let self else { return }
+
+            self.presenter?.hideLoadingIndicator()
+
+            switch result {
+            case let .success(paymentMethods):
+                self.presentComponent(with: paymentMethods)
+
+            case let .failure(error):
+                self.presentAlert(with: error)
+            }
+        }
+    }
+
+    // MARK: Presentation
+
+    internal func presentComponent(with paymentMethods: PaymentMethods) {
+        do {
+            let component = try applePayComponent(from: paymentMethods)
+            guard let componentViewController = component?.viewController else { return }
+            presenter?.present(viewController: componentViewController, completion: nil)
+            applePayComponent = component
+        } catch {
+            self.presentAlert(with: error)
+        }
+    }
+
+    internal func applePayComponent(from paymentMethods: PaymentMethods?) throws -> ApplePayComponent? {
+        guard
+            let paymentMethod = paymentMethods?.paymentMethod(ofType: ApplePayPaymentMethod.self),
+            let applePayPayment = try? ApplePayPayment(payment: ConfigurationConstants.current.payment,
+                                                       brand: ConfigurationConstants.appName)
+        else { throw IntegrationError.paymentMethodNotAvailable(paymentMethod: CardPaymentMethod.self)
```

Comment:
```suggestion
        else { throw IntegrationError.paymentMethodNotAvailable(paymentMethod: ApplePayPaymentMethod.self)
```

Reply:
Ah!. I missed this while copying the error :(. 434daaef01d20f3f62437a7110b7baf93ccfb13f